<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>516</width>
    <height>406</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Naming Convention Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QLabel" name="uiBanner_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>50</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 75 bold italic 18pt &quot;Trebuchet MS&quot;;
color: rgb(220, 220, 220);
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(20, 20, 20, 255), stop:1 rgba(40, 40, 40, 255));</string>
      </property>
      <property name="text">
       <string>Naming Convention Editor</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="margin">
       <number>9</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="contents_frame">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="list_layout">
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="selector_layout">
           <item>
            <widget class="QRadioButton" name="rules_radioButton">
             <property name="text">
              <string>Rules</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="tokens_radioButton">
             <property name="text">
              <string>Tokens</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="filter_layout">
           <property name="spacing">
            <number>1</number>
           </property>
           <item>
            <widget class="QLineEdit" name="filter_lineEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="placeholderText">
              <string>Filter by name...</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="add_button">
             <property name="minimumSize">
              <size>
               <width>22</width>
               <height>22</height>
              </size>
             </property>
             <property name="text">
              <string>+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="remove_button">
             <property name="minimumSize">
              <size>
               <width>22</width>
               <height>22</height>
              </size>
             </property>
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QListWidget" name="items_listWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="properties_layout">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QFrame" name="rules_frame">
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="expr_label">
              <property name="text">
               <string>Expression</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="expr_layout">
              <property name="spacing">
               <number>1</number>
              </property>
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLineEdit" name="expr_lineEdit">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>215</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>16777215</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="expr_button">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>22</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="text">
                 <string>&lt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="tokens_frame">
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QTableWidget" name="values_tableWidget">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>150</height>
               </size>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
              <property name="cornerButtonEnabled">
               <bool>false</bool>
              </property>
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>140</number>
              </attribute>
              <attribute name="horizontalHeaderMinimumSectionSize">
               <number>22</number>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <attribute name="verticalHeaderDefaultSectionSize">
               <number>22</number>
              </attribute>
              <attribute name="verticalHeaderHighlightSections">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Key</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Value</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="padding_frame">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="padding_label">
                 <property name="text">
                  <string>Padding</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="padding_spinBox">
                 <property name="value">
                  <number>3</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="default_groupBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>295</width>
                <height>0</height>
               </size>
              </property>
              <property name="title">
               <string>Default Value</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QLineEdit" name="default_lineEdit"/>
               </item>
               <item>
                <widget class="QSpinBox" name="default_spinBox"/>
               </item>
               <item>
                <widget class="QComboBox" name="default_comboBox"/>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>rules_radioButton</tabstop>
  <tabstop>tokens_radioButton</tabstop>
  <tabstop>filter_lineEdit</tabstop>
  <tabstop>add_button</tabstop>
  <tabstop>remove_button</tabstop>
  <tabstop>items_listWidget</tabstop>
  <tabstop>expr_lineEdit</tabstop>
  <tabstop>expr_button</tabstop>
  <tabstop>values_tableWidget</tabstop>
  <tabstop>default_groupBox</tabstop>
  <tabstop>default_lineEdit</tabstop>
  <tabstop>default_spinBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>rules_radioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>list_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>441</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tokens_radioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>list_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>155</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>83</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>items_listWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>list_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filter_lineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>filter_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>add_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>remove_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>remove_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>437</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>expr_lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>rule_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>padding_spinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>token_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>default_groupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>token_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>518</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>default_lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>token_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>305</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>default_spinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>token_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>410</x>
     <y>340</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>default_comboBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>token_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>519</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>list_clicked()</slot>
  <slot>list_changed()</slot>
  <slot>filter_changed()</slot>
  <slot>token_changed()</slot>
  <slot>add_clicked()</slot>
  <slot>remove_clicked()</slot>
  <slot>rule_updated()</slot>
  <slot>token_updated()</slot>
 </slots>
</ui>
